apps/nsq_to_file/file_logger.go

```diff
@@ -278,3 +329,35 @@ func (f *FileLogger) updateFile() {
 		f.writer = f
 	}
 }
+
+// makeOutputFilename translates a file path from work dir to output dir
+func makeOutputPath(workPath string, workDir string, outputDir string) string {
+	if workDir == outputDir {
+		return workPath
+	}
+	return filepath.Join(outputDir, strings.TrimPrefix(workPath, workDir))
+}
+
+func makeOutputDir(path string) {
+	dir, _ := filepath.Split(path)
+	if dir != "" {
+		err := os.MkdirAll(dir, 0770)
+		if err != nil {
+			log.Fatalf("ERROR: %s Unable to create %s", err, dir)
+		}
+	}
+}
+
+func atomicRename(src, dst string) error {
```

Comment:
I might have called this `exclusiveRename()` because the file does exist in two places for a moment, and because it emulates the `open(..., O_EXCL)`, but this is also fine.